linux/iscsi.go

```diff
@@ -220,11 +221,11 @@ func loginToTarget(targets model.IscsiTargets, targetIqn string, ifaces []*model
 			log.Debug("Found target :", "Target:", targetIqn)
 			// Update Chap credentials
 			if chapUser != "" && chapPassword != "" {
-				err = updateChapUser(target, chapUser)
+				err = updateChapUser(target, targets, chapUser)
```

Comment:
Raunak, looks like we are looping through each target:portal pair for chap and connection method too. Can we just change addTarget() alone to login to each target:portal pair instead and not loop within each function?

Reply:
Sure. The addTarget method was already overloaded with/without ifaces. Also connection method looks to be updated after the login. Any repercussions of any of these if we preset it before login?

Reply:
I think we just need to add portal here within addTarget()
```
args := []string{"--mode", "node", "--targetname", target.Name, "--login"}
```
and remove break after each login and don't error out if single target login fails. So any login fails we don't update connection method for that target:portal pair alone and we might update during next volume stage.

Also, here we need to consider CHAP being toggled. i.e if chap username/password is empty, but previously they were applied, we need to reset with each session too.